examples_tests/31.SkinningDataBenchmark/main.cpp

```diff
@@ -4,396 +4,385 @@
+
+    auto* am = device->getAssetManager();
+    video::IVideoDriver* driver = device->getVideoDriver();
+
+    IAssetLoader::SAssetLoadParams lp;
+    auto vertexShaderBundle = am->getAsset("../2.vert", lp);
+    auto vertexShaderBundle_3 = am->getAsset("../3.vert", lp);
+    auto fragShaderBundle = am->getAsset("../basic.frag", lp);
+    ICPUSpecializedShader* shaders[2][2];
+    shaders[0][0] = IAsset::castDown<ICPUSpecializedShader>(vertexShaderBundle.getContents().begin()->get());
+    shaders[0][1] = IAsset::castDown<ICPUSpecializedShader>(fragShaderBundle.getContents().begin()->get());
+    shaders[1][0] = IAsset::castDown<ICPUSpecializedShader>(vertexShaderBundle_3.getContents().begin()->get());
+    shaders[1][1] = IAsset::castDown<ICPUSpecializedShader>(fragShaderBundle.getContents().begin()->get());
+
+    auto createMeshBufferFromGeometryCreatorReturnData = [&](asset::IGeometryCreator::return_type& geometryObject, ICPUMeshBuffer* meshBuffer, uint16_t boneID)
     {
-        for (size_t j = 0u; j < 4u; ++j)
-            memcpy(mat3x4 + 4*j, _src + 4*7*i + 3*j, 3*sizeof(float));
-        for (size_t j = 0u; j < 3u; ++j)
-            memcpy(mat3 + 4*j, _src + 4*7*i + size_mat3x4_src + 3*j, 3*sizeof(float));
+        asset::SBlendParams blendParams;
+        asset::SRasterizationParams rasterParams;
+        rasterParams.faceCullingMode = asset::EFCM_NONE;
+
+        for (int i = 0; i < SVertexInputParams::MAX_ATTR_BUF_BINDING_COUNT; i++)
+            meshBuffer->setVertexBufferBinding(std::move(geometryObject.bindings[i]), i);
+        meshBuffer->setBoundingBox(geometryObject.bbox);
+        meshBuffer->setIndexCount(geometryObject.indexCount);
+        meshBuffer->setIndexType(geometryObject.indexType);
+        meshBuffer->setIndexBufferBinding(std::move(geometryObject.indexBuffer));
+        
+        auto newInputParams = geometryObject.inputParams;
+        newInputParams.enabledAttribFlags |= 0b10000u;
+        newInputParams.enabledBindingFlags |= 0b10u;
+        newInputParams.bindings[1].inputRate = EVIR_PER_VERTEX;
+        newInputParams.bindings[1].stride = 0u;
+        newInputParams.attributes[4].binding = 1u;
+        newInputParams.attributes[4].format = EF_R16_UINT;
+        newInputParams.attributes[4].relativeOffset = 0u;
+
+        SBufferBinding<ICPUBuffer> boneIDBuffer;
+        boneIDBuffer.offset = 0u;
+        boneIDBuffer.buffer = core::make_smart_refctd_ptr<ICPUBuffer>(meshBuffer->calcVertexCount() * sizeof(uint16_t));
+
+        uint16_t* buffPtr = static_cast<uint16_t*>(boneIDBuffer.buffer->getPointer());
+        for (int i = 0; i < meshBuffer->calcVertexCount(); i++)
+            *(buffPtr + i) = boneID;
+
+
+        meshBuffer->setVertexBufferBinding(std::move(boneIDBuffer), 1);
+
+        auto pipeline = core::make_smart_refctd_ptr<ICPURenderpassIndependentPipeline>(nullptr, nullptr, nullptr, newInputParams, blendParams, geometryObject.assemblyParams, rasterParams);
```

Comment:
arent you missing pipeline layout?

Reply:
definitely looks like you are @Przemog1

Reply:
no need, these mesh buffers are inputs to mesh packer